modules/editor/message/PrimaryContentEditor.tsx

```diff
@@ -25,28 +25,32 @@ export function PrimaryContentEditor(props: PrimaryContentEditorProps) {
         error={form.field("content").error}
         {...form.field("content").inputProps}
       />
-      <RowContainer>
-        <InputField
-          id={`_${message.id}_username`}
-          label="Username"
-          maxLength={80}
-          error={form.field("username").error}
-          {...form.field("username").inputProps}
-        />
-        <InputField
-          id={`_${message.id}_avatar`}
-          label="Avatar URL"
-          error={form.field("avatar").error}
-          {...form.field("avatar").inputProps}
-        />
-      </RowContainer>
-      <FileInputField
-        id={`_${message.id}_files`}
-        label="Files"
-        maxSize={8 * 1024 ** 2}
-        value={message.files}
-        onChange={files => message.set("files", files)}
-      />
+      {!message.url && ( // hide these fields since it isn't editable anymore
```

Comment:
Disable them instead

Reply:
I think it's worth to hide or create setting to do this. If there's a lot messages useless disabled fields take a huge amount of space. On mobile these fields even take more height than main content.

Reply:
Hiding the options, especially abruptly like this, isn't good.

Reply:
Btw there's no disable prop for file input

Reply:
I think hiding with animation & notice about readonly fields will satisfy anyone

Reply:
It may satisfy, but won't give the best possible user experience. I'll work on adding a disable prop in a minute.

Reply:
3 disabled fields takes REALLY a lot of place. Like 1/4 of screenspace on my Pixel 3. For EACH message.
Consider something like this
![image](https://user-images.githubusercontent.com/16510197/108422048-18d83880-7247-11eb-9da2-7460a0536517.png)

Reply:
Just pushed a disabled prop for the file input

Reply:
I'm sure that disabled fields 100% not the best possible UI/UX, especially for mobile

Reply:
Hi, didn't see your screenshots when I posted that. Maybe it's best to allow collapsing those 3 with a Section? Name it "Extras" or similar.

Reply:
Not bad alternative. It also save space for multiple unsent messages. But anyway there's should be some notice

Reply:
Try to find out if you can copy the 6000 character limit notice's logic, as it animates in and out quite pleasantly

Reply:
Limit notice with icon currently doesn't align properly.
![image](https://user-images.githubusercontent.com/16510197/108431060-69ee2980-7253-11eb-9dac-6a46a44d20a8.png)

Also when hidden they take extra free space and doesn't allow sections to justify with equal gaps.
![image](https://user-images.githubusercontent.com/16510197/108431119-812d1700-7253-11eb-8604-dd594a7222f8.png)

I will add some error-like warning but further fixes is out of this PR topic

Reply:
I'm aware but haven't found a way to fix that (yet). Try a small section, so one without a border. That should hide the gap.

Reply:
Please, make a warning variation for error notice. With usual exclamation sign

Reply:
I added the `variant` prop to `InputError` just now